### PR TITLE
feat: add access token credentials

### PR DESCRIPTION
**Pull request by arithmetic1728**
(1) Add `UserAccessTokenCredentials` class which uses `gcloud auth print-access-token` to obtain user account access token. This token is used to access api endpoint when mtls is enforced.
(2) Remove `google.auth._cloud_sdk.load_authorized_user_credentials` since it is a duplicate of `google.oauth2.credentials.from_authorized_user_info`. Also has to remove this to avoid cyclic import problem.

### CODE REVIEW

File: google/auth/_cloud_sdk.py [hunk 1]
```diff
@@ -122,3 +111,42 @@ def get_project_id():
...
+                (command,)
+                + _CLOUD_SDK_USER_ACCESS_TOKEN_COMMAND
+                + ("--account=" + account,)
+            )
+        else:
+            command = (command,) + _CLOUD_SDK_USER_ACCESS_TOKEN_COMMAND
+
+        access_token = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        # remove the trailing "\n"
+        return access_token.decode("utf-8")[:-1]
```
Comment (busunkim96) [inline google/auth/_cloud_sdk.py]: [`.strip()`](https://docs.python.org/3.6/library/stdtypes.html?highlight=strip#str.strip) will remove leading and trailing whitespace

File: google/oauth2/credentials.py [hunk 2]
```diff
@@ -292,3 +293,50 @@ def to_json(self, strip=None):
             prep = {k: v for k, v in prep.items() if k not in strip}
 
         return json.dumps(prep)
+
+
+class UserAccessTokenCredentials(credentials.Credentials):
+    """Access token credentials for user account.
+
+    Obtain the access token for a given user account or the current active
+    usr account with the `gcloud auth print-access-token` command.
```
Comment (busunkim96) [inline google/oauth2/credentials.py]:
```suggestion
    user account with the ``gcloud auth print-access-token`` command.
```

File: google/auth/_cloud_sdk.py [hunk 3]
```diff
@@ -122,3 +111,42 @@ def get_project_id():
         return configuration["configuration"]["properties"]["core"]["project"]
     except KeyError:
         return None
+
+
+def get_auth_access_token(account=None):
+    """Load user access token with the `gcloud auth print-access-token` command.
```
Comment (busunkim96) [inline google/auth/_cloud_sdk.py]:
```suggestion
    """Load user access token with the ``gcloud auth print-access-token`` command.
```

File: google/oauth2/credentials.py [hunk 4]
```diff
@@ -292,3 +293,50 @@ def to_json(self, strip=None):
...
+        Args:
+            account (str): Account to get the access token for.
+
+        Returns:
+            google.oauth2.credentials.UserAccessTokenCredentials: The created
+                credentials with the given account.
+        """
+        return self.__class__(account=account)
+
+    def refresh(self, request):
```
Comment (busunkim96) [inline google/oauth2/credentials.py]: Is it possible to default this to None, or does this break something?
```suggestion
    def refresh(self, request=None):
```

File: google/auth/exceptions.py [hunk 5]
```diff
@@ -28,5 +28,9 @@ class RefreshError(GoogleAuthError):
     failed."""
 
 
+class UserAccessTokenError(GoogleAuthError):
+    """Used to indicate `gcloud auth print-access-token` command failed."""
```
Comment (busunkim96) [inline google/auth/exceptions.py]:
```suggestion
    """Used to indicate ``gcloud auth print-access-token`` command failed."""
```

File: google/auth/_cloud_sdk.py (hunk 1 above)
Comment (arithmetic1728) [reply google/auth/_cloud_sdk.py]: done

File: google/oauth2/credentials.py (hunk 2 above)
Comment (arithmetic1728) [reply google/oauth2/credentials.py]: done

File: google/auth/_cloud_sdk.py (hunk 3 above)
Comment (arithmetic1728) [reply google/auth/_cloud_sdk.py]: done

File: google/oauth2/credentials.py (hunk 4 above)
Comment (arithmetic1728) [reply google/oauth2/credentials.py]: this function override the abstract method defined in the [base class](https://github.com/googleapis/google-auth-library-python/blob/master/google/auth/credentials.py#L79), so `request` cannot be default to None.

File: google/auth/exceptions.py (hunk 5 above)
Comment (arithmetic1728) [reply google/auth/exceptions.py]: done

**Pull request merged**